### PR TITLE
omf-python: Document numpy array return values

### PR DESCRIPTION
**Pull request by jsutton24**

### CODE REVIEW

**Comment by jsutton24** [comment]
The build being broken on Rust beta can be attributed to pyo3, not our code.

File: omf-python/src/file/reader.rs [hunk 1]
```diff
@@ -388,7 +403,7 @@ impl PyReader {
         }
     }
 
-    /// Read bytes of an Image.
+    /// Read bytes of an Image and return a bytes array.
```
Comment (Tim-Evans-Seequent) [inline omf-python/src/file/reader.rs]: Could be more descriptive.
```suggestion
    /// Read the encoded bytes of an Image and return them as a bytes array. The bytes
    /// may be in JPEG or PNG format; use the image header to distinguish the two.
```

File: omf-python/src/file/reader.rs [hunk 2]
```diff
@@ -400,7 +415,10 @@ impl PyReader {
             .map(|b| PyBytes::new_bound(py, &b))
     }
 
-    /// Read a Number array.
+    /// Read a Number array and return a tuple of two numpy arrays: the values
+    /// themselves, and a mask indicating any values that are null. The number
+    /// array’s type is one of :class:`numpy.float32`, :class:`numpy.float64`,
+    /// :class:`numpy.int64`, or :class:`numpy.datetime64`.
```
Comment (Tim-Evans-Seequent) [inline omf-python/src/file/reader.rs]: I think this should document the exact dtypes that can be returned, which are IIRC `"datetime64[us]"` for timestamps and `"datetime64[D]"` for dates.

File: omf-python/src/file/reader.rs [hunk 3]
```diff
@@ -511,7 +539,10 @@ impl PyReader {
         boundaries.map_err(OmfException::py_err)
     }
 
-    /// Read a RegularSubblock array.
+    /// Read a RegularSubblock array and return a tuple of two numpy arrays, the
+    /// first containing parents (as u, v, w triplets), and the second
+    /// containing the corresponding corners (as u_min, v_min, w_min, u_max,
+    /// v_max, w_max). The type of both arrays is :class:`numpy.uint32`.
```
Comment (Tim-Evans-Seequent) [inline omf-python/src/file/reader.rs]: Minor improvements:
```suggestion
    /// Read a RegularSubblock array and return a tuple of two numpy arrays, the
    /// first containing parent block indices as (u, v, w) triplets, and the second
    /// containing the corresponding corners as (u_min, v_min, w_min, u_max,
    /// v_max, w_max). The type of both arrays is :class:`numpy.uint32`.
```

File: omf-python/src/file/reader.rs [hunk 4]
```diff
@@ -525,7 +556,10 @@ impl PyReader {
         )
     }
 
-    /// Read a FreeformSubblock array.
+    /// Read a FreeformSubblock array and return a tuple of two numpy arrays,
+    /// the first containing parents (as u, v, w triplets), and the second
+    /// containing the corresponding corners (as u_min, v_min, w_min, u_max,
+    /// v_max, w_max). The type of both arrays is :class:`numpy.uint32`.
```
Comment (Tim-Evans-Seequent) [inline omf-python/src/file/reader.rs]: The corners of a free-form sub-blocked model are float, not integer.

**Review by Tim-Evans-Seequent — APPROVED** [review]

File: omf-python/src/file/reader.rs [hunk 5]
```diff
@@ -364,15 +377,17 @@ impl PyReader {
         )
     }
 
-    /// Read a Name array.
+    /// Read a Name array and return the values as a list of optional strings.
```
Comment (andrew-catalyst) [inline omf-python/src/file/reader.rs]: It looks like they might not be optional in this case?

File: omf-python/src/geometry.rs [hunk 6]
```diff
@@ -17,11 +17,17 @@ pub struct PyPointSet(pub PointSet);
 #[pymethods]
 impl PyPointSet {
     #[getter]
+    /// Origin of the surface relative to the project coordinate reference
```
Comment (andrew-catalyst) [inline omf-python/src/geometry.rs]: Maybe make these 'Origin of the geometry relative to the project origin.' perhaps?

File: omf-python/src/geometry.rs [hunk 7]
```diff
@@ -17,11 +17,17 @@ pub struct PyPointSet(pub PointSet);
...
+    /// system.
     fn origin<'py>(&self, py: Python<'py>) -> Bound<'py, PyArray1<f64>> {
         PyArray1::from_slice_bound(py, &self.0.origin)
     }
 
     #[getter]
+    /// Array with `Vertex` type storing the vertex locations.
+    ///
+    /// Add `origin` and the project origin to get the locations relative
+    /// to the project coordinate reference system.
```
Comment (andrew-catalyst) [inline omf-python/src/geometry.rs]: Maybe for this comment 'These are the positions of the vertices relative to the element origin.' might be enough?

File: omf-python/src/geometry.rs (hunk 6 above)
Comment (jsutton24) [reply omf-python/src/geometry.rs]: Okay. Are these updated comments correct?

Comment (jsutton24) [reply omf-python/src/geometry.rs]: (These descriptions came from the existing `PySurface` docstrings, which I’ve also updated to be consistent with the requested changes.)

Comment (andrew-catalyst) [reply omf-python/src/geometry.rs]: They seem to say 'surface' I think its better to call them geometry, as they're not all surfaces.

Comment (jsutton24) [reply omf-python/src/geometry.rs]: > They seem to say 'surface' I think its better to call them geometry, as they're not all surfaces.

Oops, I’ll fix that.

File: omf-python/src/geometry.rs (hunk 7 above)
Comment (andrew-catalyst) [reply omf-python/src/geometry.rs]: IMO change 'Array with `Vertex` type storing the vertex locations' to 'Array with `Vertex` type storing  vertices'.

**Review by andrew-catalyst — APPROVED** [review]

**Pull request merged**